### ISSUE TITLE
more durable storage?

### ISSUE THREAD

**Issue by lifeofguenter** [yukimochi/Activity-Relay#67]
Would it be possible to add support for a more durable storage?

I just had a crash where all data is lost in redis. No OOM, all systems are still running (e.g. no restarts) but basically empty database. The unfortunate thing is that clients do not know that they need to "re-register", which is why redis is not so ideal, as you can not rely on it as source of truth.

**Comment by danorton** [comment]
@lifeofguenter Good point. My workaround is to run a root cron job daily (which I do with a persistent databases, anyway).

    23 3    *   *   *   cp -f /var/lib/redis/dump.rdb /root/backup/redis/dump.rdb

**Comment by thystra** [comment]
FYI from a downstream maintainer: this request is only **partially** addressed
in [thystra/Activity-Relay](https://github.com/thystra/Activity-Relay).

Stable v2.5.0 still uses Redis as the durable source of truth, but it improves
the Redis deployment and task-delivery model with:

- a dedicated packaged Redis instance with RDB persistence;
- preserved Redis data and actor identity across package upgrades;
- leased in-flight claims;
- claim renewal and acknowledgement;
- recovery of expired claims after worker or host interruption; and
- explicit at-least-once delivery semantics.

It does **not** add PostgreSQL, SQL, or another interchangeable storage backend.
We evaluated multi-backend support and have tabled it for now pending stronger
operator demand and a favorable value-for-work case, so the broader request in
this issue remains valid.

Release details:
https://github.com/thystra/Activity-Relay/releases/tag/v2.5.0
